engine/schema/src/com/cloud/host/dao/HostDao.java

```diff
@@ -103,4 +95,13 @@
     List<HostVO> listByType(Type type);
 
     HostVO findByIp(String ip);
+
+    /**
+     * This method will look for a host that is connected to the storage pool where the volume of the Snapshot is stored.
+     * <ul>
+     * <li>If the storage pool found for the volume of the snapshotId has more than a host, we will choose one randomly;
+     * <li>If no host is found, we throw a runtime exception
+     * </ul>
+     */
+    HostVO findHostConnectedToSnapshotStoragePoolToExecuteCommand(long snapshotId);
```

Comment:
can i propose a shorter name? like findHostToOperateOnSnapshot(long id);

Reply:
Sure that is a good name, I will change it.